src/app/pages/datasets/modules/encryption/components/dataset-unlock/dataset-unlock.component.ts

```diff
@@ -277,118 +143,119 @@ export class DatasetUnlockComponent implements FormConfiguration {
       },
       error: this.handleError,
     });
+  }
 
-    this.keyFileControl = entityEdit.formGroup.controls['key_file'] as UntypedFormControl;
-    this.unlockChildrenControl = entityEdit.formGroup.controls['unlock_children'] as UntypedFormControl;
+  handleError = (error: WebsocketError | Job): void => {
+    new EntityUtils().handleWsError(this, error, this.dialogService);
+  };
 
-    this.keyFileControl.valueChanges.pipe(untilDestroyed(this)).subscribe({
-      next: (hideKeyDatasets: boolean) => {
-        for (let i = 0; i < this.datasets.controls.length; i++) {
-          const datasetControls = (this.datasets.controls[i] as UntypedFormGroup).controls;
-          const controls = listFields[i];
-          const keyConfig = _.find(controls, { name: 'key' });
-          const nameTextConfig = _.find(controls, { name: 'name_text' });
+  unlockSubmit(payload: DatasetUnlockParams): void {
```

Comment:
Would be good to clean up those methods, e.g.:
1. UnlockDialogComponent can be using dialogRef to communicate that it has been called instead of relying on `parent`.
2. `unlockSubmit` and `onSubmit` have similar names.
3. Sounds like UnlockDialogComponent can be renamed to UnlockSummaryDialog or something similar.
4. `unlockSubmit` is too fat. There is some code that can be extracted into separate methods, like opening an Unlock summary dialog or parsing the summary.